documentation/docs/guide/experiments/experiment-settings.md

```diff
@@ -0,0 +1,395 @@
+
+The settings under each category are listed and described below.
+
+## General settings 
+
+### Dataset
+
+<GeneralSettingsDataset/>
+
+### Problem type
+
+<GeneralSettingsProblemType/>
+
+### Import config from YAML
+
+<GSImportConfigFromYaml/>
+
+### Experiment name
+
+<GSExperimentName/>
+
+### LLM backbone
+
+<GSLLMBackbone/>
+
+## Dataset settings
+
+### Train dataframe
+
+<DSTrainDataframe/>
+
+### Validation strategy
+
+<DSvalidationStrategy/>
+
+### Validation size
+
+<DSvalidationSize/>
+
+### Data sample
+
+<DSdataSample/>
+
+### Prompt column
+
+<DSpromptColumn/>
+
+### Answer column
+
+<DSanswerColumn/>
+
+### Parent ID column
+
+<DSparentIdColumn/>
+
+### Text prompt start
+
+<DStextPromptStart/>
+
+### Text answer separator
+
+<DStextAnswerSeparator/>
+
+### Add EOS token to prompt
+
+<DSaddEosTokentoprompt/>
+
+### Add EOS token to answer
+
+<DSaddEosTokentoanswer/>
+
+### Mask prompt labels
+
+<DSmaskPromptlabels/>
+
+## Tokenizer settings
+
+### Max length prompt
+
+<TSmaxLengthPrompt/>
+
+### Max length answer
+
+<TSmaxLengthAnswer/>
+
+### Max length 
+
+<TSmaxLength/>
+
+### Add prompt answer tokens
+
+<TSaddpromptanswertokens/>
+
+### Padding quantile
+
+<TSpaddingQuantile/>
+
+### Use fast
+
+Whether or not to use a Fast tokenizer if possible. Some LLM backbones only offer certain types of tokenizers and changing this setting might be needed.
+
+## Architecture settings
+
+### Backbone Dtype
+
+<ASBackboneDtype/>
+
+### Gradient Checkpointing
+
+<ASGradientcheckpointing/>
+
+### Force Embedding Gradients
+
+<ASforceEmbeddingGradients/>
+
+### Intermediate dropout
+
+<ASintermediateDropout/>
+
+### Pretrained weights
+
+<ASpretrainedWeights/>
+
+## Training settings
+
+### Optimizer
+
+Defines the algorithm or method (optimizer) to use for model training. The selected algorithm or method defines how the model should change the attributes of the neural network, such as weights and learning rate. Optimizers solve optimization problems and make more accurate updates to attributes to reduce learning losses.
+
+
+Options
+ - **Adadelta**
+    -  To learn about Adadelta, see <a href="https://arxiv.org/abs/1212.5701" target="_blank" >ADADELTA: An Adaptive Learning Rate Method</a>. 
+ - **Adam**
+    - To learn about Adam, see <a href="https://arxiv.org/abs/1412.6980" target="_blank" >Adam: A Method for Stochastic Optimization</a>. 
+ - **AdamW**
+    - To learn about AdamW, see <a href="https://arxiv.org/abs/1711.05101" target="_blank" >Decoupled Weight Decay Regularization</a>.
+ - **AdamW8bit**
+    - To learn about AdamW, see <a href="https://arxiv.org/abs/1711.05101" target="_blank" >Decoupled Weight Decay Regularization</a>.
+ - **RMSprop** 
+    - To learn about RMSprop, see <a href="https://www.cs.toronto.edu/~tijmen/csc321/slides/lecture_slides_lec6.pdf" target="_blank" >Neural Networks for Machine Learning</a>.
+ - **SGD** 
+    - H2O LLM Studio uses a stochastic gradient descent optimizer.
+
```

Comment:
Why are these not take from the mdx file?
There are a few of these.

Reply:
CC: @oshi98